### ISSUE TITLE
Inconsistent state when bubbling up from a recursive call

### ISSUE THREAD

**Issue by Fidget-Spinner** [pylbbv/pylbbv#30]
Now `help(1)` fails with a different error (a broken assertion). This is because of an inconsistency of states when entering and leaving a recursive function. Basically the following happens:

```
1st call to f (lasti saved as the tier 1 instruction)
2nd call to f (recursive)
...
nth call to f(lasti saved as the tier 2 instruction)
... bubbles back up
leaves 2nd call to f
leaves 1st call to f (lasti now out of sync since it is tier 1 instruction but actually tier 2 instruction is already available)
```

**Comment by JuliaPoo** [comment]
Possible minimum repro
```py
def f(n):
    n+n # force specialisation
    if n==0: return 0
    if n%2:
        return f(n-1)
    return f(n-1)

for _ in range(64):
    f(3)
```

**Comment by JuliaPoo** [comment]
Potential minimal repro
```py
def f(x):
    if x == 0: return 0
    if x%2: return f(x-1)
    return f(x-1)

for i in range(15):
    f(3)

f(3) # Crashes on last return 0
```

**Comment by JuliaPoo** [comment]
Fix in #34 for #33 might partially fix this issue, not sure but will merge to fix #33 for now.

**Comment by Fidget-Spinner** [comment]
Pretty sure this is completely fixed now.
